### PR TITLE
Fix DST Bugs and Data Rounding

### PR DESCRIPTION
**Pull request by benjamin-shen**
### Summary 

This pull request fixes time-related bugs and moves rounding to the backend.

- [x] Fixed Daylight Savings bugs
- [x] Moved rounding to the backend
- [x] Implemented option to round to the nearest X:15/X:45

### Test Plan 

To test the export function, run `npm run shell` in the `functions` folder and use it as a Google Cloud Functions emulator (comment out the code in `Export.vue`).

### CODE REVIEW

File: functions/src/index.ts [hunk 1]
```diff
@@ -10,20 +10,20 @@ admin.initializeApp({
...
-    if (!data.id || !data.startDate || !data.endDate) {
+    if (!data.id || !data.startDate || !data.endDate || !data.offset) {
         throw new functions.https.HttpsError('invalid-argument', 'ID missing!');
     }
-    const startDate = new Date(data.startDate);
-    const endDate = new Date(data.endDate);
     const id = data.id;
-    return getData(id, startDate, endDate, data.offset);
+    const startDate = new Date(new Date(data.startDate));
+    const endDate = new Date(new Date(data.endDate));
```
Comment (kungpaogao) [inline functions/src/index.ts]: what's the point of this?

File: functions/src/index.ts [hunk 2]
```diff
@@ -34,14 +34,17 @@ async function getData(gymName: string, startDate: Date, endDate: Date, offset:
...
+    for (const i = new Date(startDate); i < endDate; i.setTime(i.getTime() + 24 * 60 * 60 * 1000)) {
+        dates.push(new Date(i)); // UTC
     }
-    const dateHeader = dates.map(d => d.toLocaleString("en-US", { // local time string
-        weekday: 'short',
-        month: '2-digit',
-        day: '2-digit'
-    }));
+    const dateHeader = dates.map(d => {
+        return d.toLocaleString("en-US", { // UTC date string
```
Comment (kungpaogao) [inline functions/src/index.ts]: this isn't a big deal, but I think you could get rid of the brackets and `return` statement:

```typescript
... = dates.map(d =>
  d.toLocaleString( 
    ...
  )
);
```

File: functions/src/index.ts [hunk 3]
```diff
@@ -75,33 +79,35 @@ async function getData(gymName: string, startDate: Date, endDate: Date, offset:
...
     for (const time of times) {
-        const hmDate = new Date(dates[0].getTime() + (time + offset) * 60000); // local time
+        const hmDate = new Date(dates[0].getTime() + time * 60000);
         const cardioRow = [hmDate.toLocaleString("en-US", {
             hour: "numeric",
             minute: "numeric",
-            hour12: true
+            hour12: true,
+            timeZone: 'UTC'
         })]; const weightsRow = [hmDate.toLocaleString("en-US", {
```
Comment (kungpaogao) [inline functions/src/index.ts]: add line break after `;`

File: functions/src/index.ts (hunk 2 above)
Comment (benjamin-shen) [reply functions/src/index.ts]: haha I was experimenting and forgot to change it back, good catch

File: functions/src/index.ts (hunk 1 above)
Comment (benjamin-shen) [reply functions/src/index.ts]: i believe it's a mistake, I will change it back to what it was before

**Pull request merged**